public/app/core/directives/value_select_dropdown.ts

```diff
@@ -296,6 +301,40 @@ export function valueSelectDropdown($compile, $window, $timeout, $rootScope) {
         scope
       );
 
+      scope.$on('$locationChangeStart', (event, next) => {
```

Comment:
This feels very wrong to me, that this input component should listen for changes like this.

Reply:
Sorry, I'm not familiar with redux, if there are good example in Grafana, would you tell me?

Reply:
https://github.com/grafana/grafana/blob/master/public/app/core/services/bridge_srv.ts#L41-L56
Seems related code.

Reply:
And this?
https://github.com/grafana/grafana/blob/master/style_guides/redux.md